lib/pbench/server/api/resources/__init__.py

```diff
@@ -648,10 +660,20 @@ def _validate_query_params(self, request: Request, schema: Schema) -> JSONOBJECT
         parameters into a JSON object and validates them against a specified
         schema.
 
+        Note that a multi-valued query parameter can be specified *either* by
+        a comma-separated single string value *or* by a list of individual
+        values, not both.
```

Comment:
So, if the schema defines `listparm` as a list parameter, and the request provides
```
?listparm=a,b,c&listparm=d`
```
I can imagine any of the following as being "right" answers:
- `["a", "b", "c"]` (first value wins)
- `["d"]` (last values wins)
- `["a", "b", "c", "d"]` (concatenated lists)
- `RepeatedQueryParam` (...well, yeah...)

But, what I think we get is `["a,b,c", "d"]` which I think is legal (for some twisted interpretation of that) but is unlikely to be what the user wanted.

Do we just ignore this case, with the expectation that the resulting key error for `"a,b,c"` will offer enough of a clue about what the requestor did wrong?

Reply:
> Do we just ignore this case, with the expectation that the resulting key error for "a,b,c" will offer enough of a clue about what the requestor did wrong?

Yeah, that was more or less my intent, just to keep this simple. But when you repeat my words back to me like that, I don't like what I'm hearing at all, and now you're going to make me go back and massively refactor this.

I think that the concatenated list makes the most sense, but my string to list converter needs to be more general and therefore more controlled. I talked myself out of doing this, barely, and once again you've pushed me over the edge.

I'm actually not sure whether to thank you, though as usual I suspect I'll be happier with the results. 😒